ovs/datapath_test.go

```diff
@@ -0,0 +1,644 @@
+// Copyright 2017 DigitalOcean.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ovs
+
+import (
+	"errors"
+	"testing"
+)
+
+// MockOvsCLI is mocking the "ovs-dpctl (Open vSwitch) 2.10.7" version
+type MockOvsCLI struct {
+	Version uint8
+}
+
+func (cli *MockOvsCLI) Exec(args ...string) ([]byte, error) {
+	cmd := args[0]
+	switch cmd {
+	case "--version":
+		return []byte("ovs-dpctl (Open vSwitch) 2.10.7"), nil
+	case "dump-dps":
+		return []byte("system@ovs-system\nsystem@dp-test"), nil
+	case "add-dp":
+		if args[1] == "ovs-system" {
+			return []byte{}, errors.New("datapath already exists")
+
+		}
+		return []byte{}, nil
+	case "del-dp":
+		return []byte{}, nil
+	case "ct-get-limits":
+		if cli.Version < 10 {
+			return []byte{}, errors.New("ovs-dpctl: unknown command 'ct-get-limits'; use --help for help")
+		}
+		out := "default limit=0\nzone=2,limit=1000000,count=0\nzone=3,limit=1000000,count=0"
+		return []byte(out), nil
+	case "ct-set-limits":
+		return []byte{}, nil
+	case "ct-del-limits":
```

Comment:
is this command should be guarded by a version too ?

Reply:
yes I could